src/rvpt/window.h

```diff
@@ -22,12 +40,23 @@ class Window
     explicit Window(Settings settings);
     ~Window();
 
+    void add_key_callback(std::function<void(int keycode, Action action)> callback);
+    void add_mouse_callback(
+        std::function<void(float x, float y, Mouse button, Action action)> callback);
```

Comment:
Is there a reason the mouse button and mouse move are in the same callback?

I think that they should be seperate, as someone who wants to know when buttons are clicked only care about 1/1000th of combined mouse callbacks, which means doing a lot of extra calls for no reason.

Reply:
Thanks for pointing this out.

I believe I addressed these in the latest commit. Would appreciate another PR review.

Reply:
@LegendWasTaken rebase it and I'll take a look

Reply:
This change looks good. Marking this issue as resolved.